rhombus-lib/rhombus/private/amalgam/closeable.rhm

```diff
@@ -0,0 +1,40 @@
+#lang rhombus/private/amalgam/core
+
+import:
+  "core-meta.rkt" open
+  "closeable.rkt" open
+
+export:
+  Closeable
+
+namespace Closeable:
+  export:
+    close
+    rename cdef as def  
+
+  fun close(v :: Closeable):
+    v.close()
+
+  defn.sequence_macro 'cdef $(bind :: Identifier) = $(rhs :: expr_meta.Parsed)
+                       $body
+                       ...':
+    ~op_stx: self
+    def rhs_statinfo = statinfo_meta.gather(rhs)
+    def resource = Syntax.make_temp_id("resource")
+    def resource_with_statinfo = statinfo_meta.wrap(resource, rhs_statinfo)
+    values(
+      '
+        block:
+          def mutable $resource = #false
+          try:
+            ~initially:
+              $resource := ($rhs $('::'.relocate(self)) Closeable)
+            block:
+              let $bind = $resource_with_statinfo
+              #void
```

Comment:
1. Should `Closeable.def` *not* add the `#void`, so that empty body results in an error?
2. I think `Closeable.let` may be a better name here to indicate the scope.

Reply:
I agree about the name `Closeable.let`. The original motivation for adding `#void` was to allow just `Closeable.def` (or `Closeable.let`) by itself in a module body, but maybe that wasn't a useful goal.

Reply:
Both sound good.  I've changed it.